components/cli/ballerina/bal_executor.go

```diff
@@ -20,19 +20,27 @@ package ballerina
 
 import (
 	"bufio"
+	"encoding/json"
 	"fmt"
 	"os"
 	"os/exec"
 	"strings"
+	"sync"
 
 	"github.com/cellery-io/sdk/components/cli/pkg/constants"
+	"github.com/cellery-io/sdk/components/cli/pkg/image"
 	"github.com/cellery-io/sdk/components/cli/pkg/util"
 )
 
 const ballerina = "ballerina"
+const celleryEnvVar = "cellery_env_"
+const celleryImageDirEnvVar = "CELLERY_IMAGE_DIR"
 
 type BalExecutor interface {
 	Build(fileName string, iName []byte) error
+	Run(imageDir string, instanceName string, runningNode *DependencyTreeNode,
```

Comment:
Does BalExecutor abstraction need to worry about cellery related configurations? like running node, shareDependencies flag?

Reply:
Updated interface function method signature to be generic without cellery related configurations.